### PR TITLE
AWS: do not list non-iceberg table in Glue

### PR DESCRIPTION
**Pull request by jackye1995**
People might store non-Iceberg table in the same Glue database, and `listTables` should list non-Iceberg tables.

### CODE REVIEW

File: aws/src/test/java/org/apache/iceberg/aws/glue/GlueCatalogTest.java
Comment (rdblue) [inline aws/src/test/java/org/apache/iceberg/aws/glue/GlueCatalogTest.java:352]: I think we should also test the case where the namespace is non-empty, but there is no Iceberg table.

**Comment by rdblue** [comment]
Looks like a correct implementation. I think this needs a test for the non-empty drop behavior so that we know what happens there, but otherwise I think this is good. Thanks, @jackye1995.

File: aws/src/test/java/org/apache/iceberg/aws/glue/GlueCatalogTest.java
Comment (jackye1995) [reply aws/src/test/java/org/apache/iceberg/aws/glue/GlueCatalogTest.java:352]: Great catch, dropping namespace with non-empty non-Iceberg table is interesting, in theory Iceberg should not care about it, but in practice we should probably not delete the namespace. So I updated the logic to check for iceberg or non-iceberg tables, and throw exception with different message for different cases. I have updated both the unit test and also the integration test.

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java
Comment (yyanyy) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java:354]: Probably not really need to differentiate the two, but I don't think it's worth blocking the PR.

**Review by yyanyy — APPROVED** [review]
LGTM, wondering if we want to mention in AWS documentation page that iceberg table has to contain this table type property to be considered as iceberg table, in case there are other implementations to this.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks good, I'll merge. Thanks @jackye1995!

**Pull request merged**
